attestation-container/Dockerfile

```diff
@@ -13,6 +13,10 @@ COPY go.mod go.sum ./
 RUN go mod download && go mod verify
 
 COPY . .
+
+# Pre-compile tests to avoid I/O errors in ACI
+RUN go test -c && cd attest && go test -c && cd ..
```

Comment:
This is related to the workaround for the ACI IO error. 
[Using external volume](https://github.com/microsoft/CCF/pull/4860) worked when we run only unit test on ACI. But now E2E test is run on ACI as well. Using external volume avoids the IO error but somehow `go test` command is stuck while downloading go packages for E2E test.

It's hard to find out the cause because even `go test -x` is not verbose enough.

So I just instead compile the test code in the docker file. Considering test executables are not necessary in the production, I want to delete them after IO error is resolved.

Reply:
We could also add a build arg (defaults to false) to the Dockerfile to decide whether the test are built in the image or not based on that?

Reply:
That's a nice idea! Made that change in [8ebd3a7](https://github.com/microsoft/CCF/pull/4858/commits/8ebd3a7c6cbe7c896f03714d6449423bfc6c1103)